pkg/manifests/manifests.go

```diff
@@ -1404,39 +1408,77 @@ func (f *Factory) PrometheusUserWorkload(grpcTLS *v1.Secret) (*monv1.Prometheus,
 	if err != nil {
 		return nil, err
 	}
+	if f.config.UserWorkloadConfiguration.Prometheus.LogLevel != "" {
+		p.Spec.LogLevel = f.config.UserWorkloadConfiguration.Prometheus.LogLevel
+	}
 
-	if f.config.PrometheusUserWorkloadConfig.Retention != "" {
-		p.Spec.Retention = f.config.PrometheusUserWorkloadConfig.Retention
+	if f.config.UserWorkloadConfiguration.Prometheus.Retention != "" {
+		p.Spec.Retention = f.config.UserWorkloadConfiguration.Prometheus.Retention
 	}
 
 	p.Spec.Image = &f.config.Images.Prometheus
 
-	if f.config.PrometheusUserWorkloadConfig.Resources != nil {
-		p.Spec.Resources = *f.config.PrometheusUserWorkloadConfig.Resources
+	if f.config.UserWorkloadConfiguration.Prometheus.Resources != nil {
+		p.Spec.Resources = *f.config.UserWorkloadConfiguration.Prometheus.Resources
 	}
 
-	if f.config.PrometheusUserWorkloadConfig.NodeSelector != nil {
-		p.Spec.NodeSelector = f.config.PrometheusUserWorkloadConfig.NodeSelector
+	if f.config.UserWorkloadConfiguration.Prometheus.NodeSelector != nil {
+		p.Spec.NodeSelector = f.config.UserWorkloadConfiguration.Prometheus.NodeSelector
 	}
 
-	if len(f.config.PrometheusUserWorkloadConfig.Tolerations) > 0 {
-		p.Spec.Tolerations = f.config.PrometheusUserWorkloadConfig.Tolerations
+	if len(f.config.UserWorkloadConfiguration.Prometheus.Tolerations) > 0 {
+		p.Spec.Tolerations = f.config.UserWorkloadConfiguration.Prometheus.Tolerations
 	}
 
-	if f.config.PrometheusUserWorkloadConfig.ExternalLabels != nil {
-		p.Spec.ExternalLabels = f.config.PrometheusUserWorkloadConfig.ExternalLabels
+	if f.config.UserWorkloadConfiguration.Prometheus.ExternalLabels != nil {
+		p.Spec.ExternalLabels = f.config.UserWorkloadConfiguration.Prometheus.ExternalLabels
 	}
 
-	if f.config.PrometheusUserWorkloadConfig.VolumeClaimTemplate != nil {
+	if f.config.UserWorkloadConfiguration.Prometheus.VolumeClaimTemplate != nil {
 		p.Spec.Storage = &monv1.StorageSpec{
-			VolumeClaimTemplate: *f.config.PrometheusUserWorkloadConfig.VolumeClaimTemplate,
+			VolumeClaimTemplate: *f.config.UserWorkloadConfiguration.Prometheus.VolumeClaimTemplate,
 		}
 	}
 
-	if len(f.config.PrometheusUserWorkloadConfig.RemoteWrite) > 0 {
-		p.Spec.RemoteWrite = f.config.PrometheusUserWorkloadConfig.RemoteWrite
+	if len(f.config.UserWorkloadConfiguration.Prometheus.RemoteWrite) > 0 {
+		p.Spec.RemoteWrite = f.config.UserWorkloadConfiguration.Prometheus.RemoteWrite
+	}
+
+	// TODO: remove after 4.7
+
+	if f.config.ClusterMonitoringConfiguration.PrometheusUserWorkloadConfig.LogLevel != "" {
```

Comment:
When you have config fields defined both in `f.config.ClusterMonitoringConfiguration`  and `f.config.UserWorkloadConfiguration` then the former will take precedence. Couldn't this cause confusion?

Reply:
techPreview is what we have to support right now primarily hence why it takes precedence here, but that is a good point to document.